src/api_routes.py

```diff
@@ -1,42 +1,96 @@
 from fastapi import FastAPI
+from fastapi.responses import FileResponse
+from fastapi import HTTPException
+import json
+import asyncio
+import os
+
 from engine import Engine
+from search import Search
+from download import Download
+
 
 app = FastAPI()
 engine = Engine()
 
 @app.post("/v1/songs")
-async def _():
+async def process_song(link):
     """
     Receives link to a song and directs it to download
 
     Returns:
         json: unique song id based on youtube url
     """
-    raise NotImplementedError()
+    timeout_download = 60
+    download = Download(link)
+
+    success = False
+    for _ in range(timeout_download):
+        if download.is_ready():
+            success = True
+            break
+        await asyncio.sleep(1)
+
+    if not success:
+        return HTTPException(status_code=504, detail="Downloading timeout")
+    
+    try:
+        song_id = download.get_name()
+    except Exception as e:
+        return HTTPException(status_code=502, detail=e)
+    try:
+        engine.enqueue(f"downloads/{song_id}/audio.mp3")
```

Comment:
```suggestion
        engine.enqueue(f"downloads/{song_id}")
```
Engine.enqueue() takes path to directory, not audio file